src/containers/Search/Search.tsx

```diff
@@ -0,0 +1,143 @@
+
+const PAST_SEARCH_STORAGE_KEY = "pastSearch";
+const MAX_PAST_SEARCHES_STORED = 20;
+
+export const Search = () => {
+  const { updateRequestStatusCall, requestStatuses, dispatch } = useRequestStatusContext();
+  const inputRef = useRef<TextInput>(null);
+  const [searchInputValue, setSearchInputValue] = useState("");
+  const [searchedProducts, setSearchedProducts] = useState<Product[]>([]);
+
+  const [shouldShowPastSearch, setShowPastSearch] = useState(false);
+  const [pastSearches, setPastSearches] = useState<string[]>([]);
+
+  const handleInputChange = (text: string) => {
+    setShowPastSearch(!!text.length);
+    setSearchInputValue(text);
+  };
+
+  const handleProductSearch = async (pastSearch?: string) => {
+    if (updateRequestStatusCall) {
+      setShowPastSearch(false);
+
+      if (pastSearch) {
+        setSearchInputValue(pastSearch);
+      }
+
+      const lowerCaseSearchValue = (pastSearch || searchInputValue)?.toLowerCase();
+      const { data } = await updateRequestStatusCall(
+        () =>
+          baseApi.getProducts({
+            "filter[name]": lowerCaseSearchValue,
+          }),
+        URLNames.products
+      );
+
+      setSearchedProducts(data.data);
+
+      if (!pastSearch) {
+        const rawPastSearches = await AsyncStorage.getItem(PAST_SEARCH_STORAGE_KEY);
+        let updatedPastSearches: string[] = [];
+        if (rawPastSearches) {
+          updatedPastSearches = JSON.parse(rawPastSearches);
+
+          if (updatedPastSearches.length >= MAX_PAST_SEARCHES_STORED) {
+            updatedPastSearches.pop();
+          }
+        }
+
+        if (updatedPastSearches.indexOf(lowerCaseSearchValue) === -1 && lowerCaseSearchValue.length) {
+          updatedPastSearches.unshift(lowerCaseSearchValue);
+        }
+
+        await AsyncStorage.setItem(PAST_SEARCH_STORAGE_KEY, JSON.stringify(updatedPastSearches));
+
+        setPastSearches(updatedPastSearches);
+      }
+    }
+  };
+
+  useEffect(() => {
+    AsyncStorage.getItem(PAST_SEARCH_STORAGE_KEY).then((stringifiedPastSearches) => {
+      if (stringifiedPastSearches) {
+        try {
+          setPastSearches(JSON.parse(stringifiedPastSearches));
+        } catch (error) {
+          console.log(error);
+        }
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    if (inputRef) {
+      inputRef.current?.focus();
+    }
+
+    return () => {
+      dispatch?.({
+        type: URLNames.products,
+        payload: {
+          status: RequestStatus.RESET,
+        },
+      });
+    };
+  }, [dispatch]);
+  return (
+    <ScrollView style={[styles.searchContainer]}>
+      <Input
+        ref={inputRef}
+        value={searchInputValue}
+        onChange={handleInputChange}
+        onStartAdormentPress={() => handleProductSearch()}
+        startAdorment={<SearchIcon />}
+        isLoading={requestStatuses?.[URLNames.products].status === RequestStatus.REQUEST}
+      />
+      {shouldShowPastSearch && (
```

Comment:
1. not the whole list of previous searches visible. List of previous searches should be visible on input focus
 
[untitled.webm](https://user-images.githubusercontent.com/36453308/213788954-d8baf0f5-9be9-4b4b-93bb-82aa8457609d.webm)
2. there is no possibility to delete items

Reply:
fixed